design/workflow.go

```diff
@@ -316,3 +340,18 @@ var WorkflowRunOutput = Type("WorkflowRunOutput", func() {
 	Field(1, "output", WorkflowOutput, "The workflow output")
 	Field(2, "value", String, "The output value set by the Workflow run")
 })
+
+// WorkflowAssignment describes the assignment between a workflow and codesets
+var WorkflowAssignment = Type("WorkflowAssignment", func() {
+	Field(1, "workflow", String, "Workflow assigned to the codeset")
+	Field(2, "listener", WorkflowListener, "The listener responsible for the assignment")
```

Comment:
We already spoke about this listener field, so this is me trying to capture here the minutes of that discussion:
* the listener concept and thus it's attributes represent a backend implementation detail that shouldn't be exposed through this API endpoint. It should be sufficient to convert the listener state (its `availability`) as the state of the assignment and leave out the other attributes. Those other attributes do not belong in a workflow assignment business logic concept. This, at least, is my viewpoint.
* the Tekton "listener" is required for codesets, but there's no indication that this concept can be generalized to cover other types of artifacts and artifact stores. In fact, the FuseML core service itself acts as the "listener", if all information about new versions of artifacts is directed at it. 
* your interpretation is that the listener contains useful information that users might need to debug issues such as workflows not being triggered on code push events, and that we should definitely include it in the assignment. We could control how/if the information is displayed e.g. by only including it in the "verbose" version of the output.

Reply:
To follow up, after I've given this topic some more consideration:
* what we refer to as a "codeset assignment" here is a concrete example of a more elaborate business logic concept that generally speaking represents the relationship between a workflow input and an artifact (codeset, model, dataset etc) concerning how a workflow run is triggered when a new input artifact version is available
* exposing the listener to the end user has some other design repercussions that you're surely aware of by now. One of them is that the workflow store API needs to be made aware of a "listener" concept. In other words, you need to model "listener" as a business logic concept exposed by the workflow store, otherwise you'll have to interact with the Tekton backend directly from the REST API service to extract the information, which is not desirable (as mentioned in https://github.com/fuseml/fuseml-core/pull/50).

To meet you half way, here's what I suggest: remove "listener" as a field, but keep some of its attributes as global assignment attributes:
* transform `available` into a global assignment `status` field. This will represent the status of the assignment, independently of how it's implemented. The workflow backend can map it to anything. In particular, for codesets, this is mapped to the Tekton listener's availability.
* transform the listener URL into the Tekton dasboard URL that gets you to the listener object (same as we do with workflow runs). This is far more valuable to the end user who wants to debug the back-end implementation than the current `listener.name` and `listener.URL` fields are.

This way, we avoid modeling "listener" as a business logic concept and still offer the means to get the information for debugging.

Reply:
I have added a new type to represent the assignment status, it is made by the status and the dashboard url.

Reply:
Perfect, thanks !